juju/juju
# Make the user manager client facade use the new user domain

Builds upon and supersedes https://github.com/juju/juju/pull/16838.

The `usermanager` facade now uses the new user domain service. At present we are double-writing all actions, so permissions (still reliant on Mongo) continue to work.

Note that the correct implementation for `UserInfo` needs a follow-up patch to correctly factor the `IncludeDisabled` argument.

Many queries in the user domain state are rewritten for brevity and correctness, and some now use a new view that joins `user` to `user_authentication`.

## QA steps

As the admin user:
- As the `admin` user: create, set a password for, enable, disable, delete a user.
- As a normal unprivileged user: verify that the operations above are not permitted.

## Links

**Jira card:** [JUJU-5353](https://warthogs.atlassian.net/browse/JUJU-5353)

[JUJU-5353]: https://warthogs.atlassian.net/browse/JUJU-5353?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
